### PR TITLE
[BugFix] Fix schema change hang when FE meets runtime exception

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
If FE got a runtime exception when handle an alter task finish request, the alter task cannot finished normally and will hang the whole alter job until timeout.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by meegoo — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java [hunk 1]
```diff
@@ -554,9 +554,8 @@ protected void runRunningJob() throws AlterCancelException {
             LOG.info("schema change tasks not finished. job: {}", jobId);
             List<AgentTask> tasks = schemaChangeBatchTask.getUnfinishedTasks(2000);
             for (AgentTask task : tasks) {
-                if (task.getFailedTimes() >= 3) {
-                    throw new AlterCancelException(
-                            "schema change task failed after try three times: " + task.getErrorMsg());
+                if (task.isFailed() || task.getFailedTimes() >= 3) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java:557]: isFailed() means only failed one time?

Comment (sevev) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java:557]: > isFailed() means only failed one time?

`isFailed()` means whether the FE handle finish alter task successful, and alter task still retry three times in BE.

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**